pkg/policyengine/PolicyDispatcher.go

```diff
@@ -167,24 +161,31 @@ func (pH *PolicyHandler) AuthorizeAndRouteConnection(connReq *event.ConnectionRe
 }
 
 func (pH *PolicyHandler) AddPeer(peer *api.Peer) {
-	_, exist := exists(pH.knownPeers, peer.Name)
-	if exist {
+	if _, exists := pH.enabledPeers[peer.Name]; exists {
 		return
 	}
-	pH.knownPeers = append(pH.knownPeers, peer.Name)
-	plog.Infof("Added Peer %+v", pH.knownPeers)
+
+	pH.enabledPeers[peer.Name] = true
+	plog.Infof("Added Peer %s", peer.Name)
 }
 
 func (pH *PolicyHandler) DeletePeer(name string) {
-	pH.loadBalancer.RemovePeerFromServiceMap(name)
-
-	index, exist := exists(pH.knownPeers, name)
-	if !exist {
+	if _, exists := pH.enabledPeers[name]; !exists {
 		return
 	}
-	pH.knownPeers = append(pH.knownPeers[:index], pH.knownPeers[index+1:]...)
-	plog.Infof("Removed Peer(%s, %d) %+v", name, index, pH.knownPeers)
 
+	pH.loadBalancer.RemovePeerFromServiceMap(name)
+
+	delete(pH.enabledPeers, name)
+	plog.Infof("Removed Peer %s", name)
+}
+
+func (pH *PolicyHandler) DisablePeer(name string) {
+	delete(pH.enabledPeers, name)
+}
+
+func (pH *PolicyHandler) EnablePeer(name string) {
```

Comment:
Maybe you need to add check if it is exsit

Reply:
The [documentation](https://pkg.go.dev/builtin#delete)  says "If m is nil or there is no such element, delete is a no-op. "